x-pack/plugins/siem/public/components/formatted_date/index.test.tsx

```diff
@@ -10,70 +10,216 @@ import moment from 'moment-timezone';
-        </KibanaConfigContext.Provider>
-      );
-      expect(wrapper.text()).toEqual(isoDateString);
+  describe('FormattedDate', () => {
+    describe('rendering', () => {
+      test('it renders against a numeric epoch', () => {
+        const wrapper = mount(
+          <KibanaConfigContext.Provider value={mockFrameworks.default_UTC}>
+            <FormattedDate fieldName="@timestamp" value={1559079339000} />
+          </KibanaConfigContext.Provider>
+        );
+        expect(wrapper.text()).toEqual('May 28, 2019 @ 21:35:39.000');
+      });
+
+      test('it renders against a string epoch', () => {
+        const wrapper = mount(
+          <KibanaConfigContext.Provider value={mockFrameworks.default_UTC}>
+            <FormattedDate fieldName="@timestamp" value={'1559079339000'} />
+          </KibanaConfigContext.Provider>
+        );
+        expect(wrapper.text()).toEqual('May 28, 2019 @ 21:35:39.000');
+      });
+
+      test('it renders against a ISO string', () => {
+        const wrapper = mount(
+          <KibanaConfigContext.Provider value={mockFrameworks.default_UTC}>
+            <FormattedDate fieldName="@timestamp" value={'2019-05-28T22:04:49.957Z'} />
+          </KibanaConfigContext.Provider>
+        );
+        expect(wrapper.text()).toEqual('May 28, 2019 @ 22:04:49.957');
+      });
+
+      test('it renders against an empty string as an empty string placeholder', () => {
+        const wrapper = mount(
+          <TestProviders>
+            <KibanaConfigContext.Provider value={mockFrameworks.default_UTC}>
+              <FormattedDate fieldName="@timestamp" value={''} />
+            </KibanaConfigContext.Provider>
+          </TestProviders>
+        );
+        expect(wrapper.text()).toEqual('(Empty String)');
+      });
+
+      test('it renders against an null as a EMPTY_VALUE', () => {
+        const wrapper = mount(
+          <TestProviders>
+            <KibanaConfigContext.Provider value={mockFrameworks.default_UTC}>
+              <FormattedDate fieldName="@timestamp" value={null} />
+            </KibanaConfigContext.Provider>
+          </TestProviders>
+        );
+        expect(wrapper.text()).toEqual(getEmptyValue());
+      });
+
+      test('it renders against an undefined as a EMPTY_VALUE', () => {
+        const wrapper = mount(
+          <TestProviders>
+            <KibanaConfigContext.Provider value={mockFrameworks.default_UTC}>
+              <FormattedDate fieldName="@timestamp" value={undefined} />
+            </KibanaConfigContext.Provider>
+          </TestProviders>
+        );
+        expect(wrapper.text()).toEqual(getEmptyValue());
+      });
+
+      test('it renders against an invalid date time as just the string its self', () => {
+        const wrapper = mount(
+          <TestProviders>
+            <KibanaConfigContext.Provider value={mockFrameworks.default_UTC}>
+              <FormattedDate fieldName="@timestamp" value={'Evan Braden Rebecca'} />
```

Comment:
The wife should always before the kid!!! ;)

Reply:
Sorted it by age now :-)